### PR TITLE
Bootstrap updates

### PR DESCRIPTION
**Pull request by Thryphore**
Testing
* Main page tables and the details page both work on mobile

### CODE REVIEW

File: frontend/src/pages/Users.tsx [hunk 1]
```diff
@@ -1,4 +1,5 @@
 import React from 'react'
+import Table from 'react-bootstrap/Table'
 import { Button } from 'react-bootstrap'
```
Comment (anthonywittig) [inline frontend/src/pages/Users.tsx:3]: Thoughts on combining lines 2 and 3?
```
import { Button, Table } from 'react-bootstrap'
```

File: frontend/src/pages/devices/List.tsx [hunk 2]
```diff
@@ -1,10 +1,12 @@
 import React from 'react'
+import Table from 'react-bootstrap/Table'
 import { Badge, Button, OverlayTrigger, Tooltip } from 'react-bootstrap'
```
Comment (anthonywittig) [inline frontend/src/pages/devices/List.tsx:3]: Same here

Comment (anthonywittig) [reply frontend/src/pages/devices/List.tsx:3]: (alphabetized, so `Tooltip` would be last)

File: frontend/src/pages/devices/List.tsx [hunk 3]
```diff
@@ -78,23 +79,23 @@ export const List = () => {
                 <tbody>
                     {entities.map(entity =>
                         <tr key={ entity.id }>
-                            <th scope="row">
+                            <th className="align-middle">
```
Comment (anthonywittig) [inline frontend/src/pages/devices/List.tsx:82]: Looks like there are some other changes in here now, what do all of these do? It might be good to include some before/after screenshots. 

It's generally frowned upon to unrelated changes to an existing PR :P

File: frontend/src/pages/devices/List.tsx [hunk 4]
```diff
@@ -1,10 +1,11 @@
 import React from 'react'
-import { Badge, Button, OverlayTrigger, Tooltip } from 'react-bootstrap'
+import { Badge, Button, OverlayTrigger, Table, ListGroup, Tooltip } from 'react-bootstrap'
```
Comment (anthonywittig) [inline frontend/src/pages/devices/List.tsx:2]: Alphabetize (there's not a good reason for this, it's just the way I've been ordering them).

File: frontend/src/pages/devices/List.tsx (hunk 3 above)
Comment (anthonywittig) [inline frontend/src/pages/devices/List.tsx:82]: I suspect we won't have any screen readers for a while, but I like sticking with the bootstrap default examples. Thoughts on keeping the `scope="row"`? (I don't think it has any visual affects.)

File: frontend/src/pages/devices/List.tsx [hunk 5]
```diff
@@ -78,23 +79,23 @@ export const List = () => {
...
-                            <th scope="row">
+                            <th className="align-middle">
                                 <Link to={"/devices/" + entity.id}>
                                     <Button variant="link">{ entity.rawDevice.name }</Button>
                                 </Link>
                             </th>
-                            <td>{ renderEntityStatus(entity) }</td>
-                            <td>{ renderEntityBatteryLevel(entity) }</td>
-                            <th scope="row">
+                            <td className="align-middle text-left">{renderEntityStatus(entity)}</td>
```
Comment (anthonywittig) [inline frontend/src/pages/devices/List.tsx:87]: Could you show what this looks like with and w/o the `text-left`?

File: frontend/src/pages/devices/List.tsx [hunk 6]
```diff
@@ -123,14 +124,15 @@ export const List = () => {
         warnings.push.apply(warnings, getLockResponsivenessWarnings(entity))
 
         if (warnings.length === 1) {
-            return <>{ warnings[0] }</>
+
+            return <ListGroup.Item className="border-0">{ warnings[0] }</ListGroup.Item>
         }
         return (
-            <ul>
+            <ListGroup className="px-0">
```
Comment (anthonywittig) [inline frontend/src/pages/devices/List.tsx:131]: Would [flush](https://getbootstrap.com/docs/4.0/components/list-group/#flush) work in place of `px-0` and `border-0`?

File: frontend/src/pages/devices/List.tsx (hunk 3 above)
Comment (Thryphore) [reply frontend/src/pages/devices/List.tsx:82]: It has no visual effects which is why I removed it. I'll put it back in.

File: frontend/src/pages/devices/List.tsx (hunk 5 above)
Comment (Thryphore) [reply frontend/src/pages/devices/List.tsx:87]: It doesn't appear to do anything, I think it was an attempt to bring the status column left more because it doesn't line up correctly. I spent a good chunk of time trying to make it better but I realized that the other columns were like this as well so I decided it was good enough.
look

File: frontend/src/pages/devices/List.tsx (hunk 6 above)
Comment (Thryphore) [reply frontend/src/pages/devices/List.tsx:131]: I was able to change 2 of them. The 3rd one gave it a weird border
image

File: frontend/src/pages/devices/List.tsx [hunk 7]
```diff
@@ -1,243 +1,277 @@
...
+        console.log(err)
+      })
+  }
+
+  const render = () => {
+    return (
+      <>
+        <div
+          className="card"
+          style={{ marginBottom: "1rem", marginTop: "1rem" }}
```
[the remark this answers is not in the document]
Comment (Thryphore) [reply frontend/src/pages/devices/List.tsx:62]: It doesn't appear to do anything, not sure why it's there. Test code I forgot about or you put it there perhaps.

File: frontend/src/pages/devices/List.tsx [hunk 8]
```diff
@@ -1,243 +1,277 @@
...
+            <th scope="col">Status</th>
+            <th scope="col">Battery</th>
+            <th scope="col">Unit</th>
+            <th scope="col">Actions</th>
+          </tr>
+        </thead>
+        <tbody>
+          {entities.map((entity) => (
+            <tr key={entity.id}>
+              <th className="align-middle" scope="row">
```
[the remark this answers is not in the document]
Comment (Thryphore) [reply frontend/src/pages/devices/List.tsx:91]: I couldn't tell if they do anything so I removed them. They may have done something before using better bootstrap stuff.

**Comment by Thryphore** [comment]
Had merge errors, this was easier than resolving them all.
https://github.com/anthonywittig/mLock/pull/37
